src/app/main.rs

```diff
@@ -460,7 +459,7 @@ where
             // 6. Strips any `l5d-server-id` that may have been received from
             //    the server, before we apply our own.
             let endpoint_stack = client_stack
-                .push(buffer::layer(MAX_IN_FLIGHT))
+                .push(buffer::layer(max_in_flight))
```

Comment:
TIOLI: I find the the ordering that we push layers in the `inbound`/`outbound` routers to be helpful in understanding what a layer should do in the stack. I think this layer makes more sense right next to the `router`, instead of having the `strip_header`s in between.

When I first started looking at this part of the proxy, I placed a lot of significance on the ordering of layers when sometimes there it may not matter. I would have been confused why we had `strip_header` layers come in between this and the `router` layer.